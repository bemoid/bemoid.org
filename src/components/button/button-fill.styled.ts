import { css } from '@emotion/react'
import styled from '@emotion/styled'

import { Button } from '.'

export const ButtonFill = styled(Button)<{
  variant: 'primary'
}>`
  ${({ variant }) =>
    variant === 'primary' &&
    css`
      color: #fff;
      background-color: #ff5f6d;

      &:hover {
        color: #fff;
        background-color: #ef4e5b;
      }
    `}
`
