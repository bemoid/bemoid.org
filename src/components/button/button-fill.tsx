import { forwardRef } from 'react'

import type { ButtonProps } from './button'

import * as Styled from './button-fill.styled'

export type ButtonFillProps = ButtonProps & {
  variant?: 'primary'
}

export const ButtonFill = forwardRef<HTMLAnchorElement, ButtonFillProps>(
  ({ variant = 'primary', children, ...rest }, ref) => {
    return (
      <Styled.ButtonFill variant={variant} ref={ref} {...rest}>
        {children}
      </Styled.ButtonFill>
    )
  }
)
