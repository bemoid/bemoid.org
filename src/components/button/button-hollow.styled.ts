import { css } from '@emotion/react'
import styled from '@emotion/styled'

import { Button } from '.'

export const ButtonHollow = styled(Button)<{
  variant: 'primary' | 'neutral'
}>`
  border: 2px solid transparent;

  ${({ variant }) =>
    variant === 'primary' &&
    css`
      color: #ff5f6d;
      border-color: #ff5f6d;

      &:hover {
        background-color: #ff5f6d;
        color: #fff;
      }
    `}

  ${({ variant }) =>
    variant === 'neutral' &&
    css`
      color: #484357;
      border-color: #ebebf2;

      &:hover {
        border-color: #dfdfe5;
      }
    `}
`
