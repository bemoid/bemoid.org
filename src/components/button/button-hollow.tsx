import { forwardRef } from 'react'

import type { ButtonProps } from './button'

import * as Styled from './button-hollow.styled'

export type ButtonHollowProps = ButtonProps & {
  variant?: 'primary' | 'neutral'
}

export const ButtonHollow = forwardRef<HTMLAnchorElement, ButtonHollowProps>(
  ({ variant = 'primary', children, ...rest }, ref) => {
    return (
      <Styled.ButtonHollow variant={variant} ref={ref} {...rest}>
        {children}
      </Styled.ButtonHollow>
    )
  }
)
