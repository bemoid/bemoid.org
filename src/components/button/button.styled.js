import { css } from '@emotion/core'
import styled from '@emotion/styled'

export const Button = styled.a`
  position: relative;
  cursor: pointer;
  display: inline-block;
  vertical-align: middle;
  line-height: inherit;
  text-decoration: none;
  outline: 0;
  text-align: center;
  font-size: 18px;
  padding: 0.5em 0.8em;
  transition: color 0.2s ease-in-out, background-color 0.2s ease-in-out, border-color 0.2s ease-in-out;

  ${({ size }) => size === 'medium' && css`
    @media (min-width: 768px) {
      padding: 0.6em 1em;
    }
  `}

  ${({ size }) => size === 'large' && css`
    @media (min-width: 768px) {
      padding: 0.8em 1.2em;
    }
  `}

  ${({ expand }) => expand && css`
    display: block;
    width: 100%;
  `}
`
