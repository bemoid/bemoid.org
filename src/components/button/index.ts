export { Button } from './button'
export { ButtonFill } from '../button/button-fill'
export { ButtonHollow } from '../button/button-hollow'
