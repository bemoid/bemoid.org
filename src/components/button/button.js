import * as Styled from './button.styled'

import { string, bool, node } from 'prop-types'

export const Button = ({ href, size, expand, className, children }) => {
  return (
    <Styled.Button href={href} size={size} expand={expand} className={className}>
      {children}
    </Styled.Button>
  )
}

Button.propTypes = {
  href: string.isRequired,
  size: string,
  expand: bool,
  className: string,
  children: node.isRequired,
}

Button.defaultProps = {
  size: 'medium',
  className: '',
  expand: false,
}
