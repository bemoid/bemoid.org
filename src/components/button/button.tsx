import { forwardRef, ReactNode } from 'react'

import * as Styled from './button.styled'

export type ButtonProps = {
  href: string
  size?: 'medium' | 'large'
  expand?: boolean
  className?: string
  children: ReactNode
  onClick?: () => void
}

export const Button = forwardRef<HTMLAnchorElement, ButtonProps>(
  (
    {
      href,
      size = 'medium',
      expand = false,
      className = '',
      children,
      onClick,
      ...rest
    },
    ref
  ) => {
    return (
      <Styled.Button
        href={href}
        size={size}
        expand={expand}
        className={className}
        onClick={onClick}
        ref={ref}
        data-testid="button"
        {...rest}
      >
        {children}
      </Styled.Button>
    )
  }
)
