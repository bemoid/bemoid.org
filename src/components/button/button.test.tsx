import { screen, render, fireEvent } from '@testing-library/react'

import { Button } from './button'

describe('Button component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Button href="https://bemoid.org/">Button</Button>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('calls `onClick` event', () => {
    const handleClick = jest.fn()

    render(
      <Button href="https://bemoid.org/" onClick={handleClick}>
        Button
      </Button>
    )

    const button = screen.getByTestId('button')

    fireEvent.click(button)
    expect(handleClick).toHaveBeenCalled()
  })
})
