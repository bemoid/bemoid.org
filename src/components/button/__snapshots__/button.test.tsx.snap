// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<DocumentFragment>
  <a
    class=" css-ybdpjn"
    data-testid="button"
    href="https://bemoid.org/"
  >
    Button
  </a>
</DocumentFragment>
`;
